ui/src/widgets/ui-chart/UIChart.vue

```diff
@@ -191,10 +194,52 @@ export default {
         }
         const chart = new Chart(el, config)
 
+        // Useful for debugging: uncomment to expose the chart object to the window
+        // window.uiChart = window.uiChart || {}
+        // window.uiChart[this.id] = chart
+
         // don't want chart to be reactive, so we can use shallowRef
         this.chart = shallowRef(chart)
+
+        // ensure the chart is updated with the correct interaction mode
+        // based on the type of chart we are creating
+        this.updateInteraction()
     },
     methods: {
+        updateInteraction () {
+            switch (this.chart.config.type) {
+            case 'line':
+                this.chart.options.interaction.axis = 'x'
+                this.chart.options.interaction.mode = 'index'
+                break
+            default:
+                delete this.chart.options.interaction.axis
+                this.chart.options.interaction.mode = 'nearest' // default
+                break
```

Comment:
I mentioned [here](https://github.com/FlowFuse/node-red-dashboard/issues/1254#issuecomment-2371774649) that `mode` should be `x` for the line charts, as if you have different lines with a different number of data points, then the tooltips get mis-aligned.

```suggestion
            case 'line':
                this.chart.options.interaction.mode = 'x'
                break
            case 'scatter':
                this.chart.options.interaction.axis = 'x'
                this.chart.options.interaction.mode = 'nearest'
                break
            default:
                delete this.chart.options.interaction.axis
                this.chart.options.interaction.mode = 'index' // default
                break
```

Having played around too, the bar and donut charts benefit from `index` type, with the scatter reverting to the `nearest` tooltip type.

Reply:
> I mentioned https://github.com/FlowFuse/node-red-dashboard/issues/1254#issuecomment-2371774649 that mode should be x for the line charts


I misread the API docs https://www.chartjs.org/docs/latest/configuration/interactions.html#modes (didnt scroll down enough to see there was a mode 'x' and thought what you entered was a typo. My bad!)